app/main.py

```diff
@@ -1,3 +1,59 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        elif isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+            return self
+        elif isinstance(other, Distance):
+            self.km += other.km
+            return self
+
+    def __mul__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+
+    def __truediv__(self, other: Distance | float | int) -> Distance:
```

Comment:
Add processing of the `Distance` class instance here as well in case it is indicated in the type annotation